### PR TITLE
Adding 10 New CodeWOF Questions

### PR DESCRIPTION
**Pull request by jqi26**
Adding new questions to CodeWOF (partly as an extra incentive to use my experimental version).

- Added 4 function questions.
- Added 2 debug questions.
- Added 2 parsons questions.
- Added 2 program questions.

### CODE REVIEW

File: codewof/programming/content/en/vertical-line/question.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#Vertical Line
+
+Write a function `is_vertical(point1, point2)` that takes two points represented by tuples of the form (x, y).
+
+It should return True if the points lie on a vertical line, or False otherwise.
```
Comment (JackMorganNZ) [inline codewof/programming/content/en/vertical-line/question.md:5]: Modify to be similar to the suggestion for the horizontal line question.

File: codewof/programming/content/structure/questions.yaml [hunk 2]
```diff
@@ -953,3 +953,91 @@ square-area:
...
+    1: normal
+    2: normal
+    3: normal
+
+first-letters:
+  types:
+    - parsons
+  parsons-extra-lines:
+    - 'result += words[0]'
+    - 'result += word[-1]'
```
Comment (JackMorganNZ) [inline codewof/programming/content/structure/questions.yaml:1023]: Remove this option, the question is hard enough as is.

File: codewof/programming/content/en/first-letters/question.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# First Letters
```
[the remark this answers is not in the document]
Comment (JackMorganNZ) [reply codewof/programming/content/en/first-letters/question.md:1]: Just seen this is a parsons  problem which makes it much easier. I think you're fine to include it, but still include the hint explaining how split works, and difficult note at the top.

**Review by JackMorganNZ — CHANGES_REQUESTED** [review]

File: codewof/programming/content/en/acceleration-calculator/question.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# Acceleration Calculator
+The following function `acceleration(velocity_initial, velocity_final, time)` takes the initial and final velocity of an object (m/s) and the time (s), and should return the correct acceleration (m/s<sup>2</sup>).
```
[the remark this answers is not in the document]
Comment (jqi26) [reply codewof/programming/content/en/acceleration-calculator/question.md:2]: Fair enough, I added an explanation of the formula.

File: codewof/programming/content/en/first-letters/question.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jqi26) [reply codewof/programming/content/en/first-letters/question.md:1]: Added a hint and warning.

File: codewof/programming/content/en/horizontal-line/question.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#Horizontal Line
+
+Write a function `is_horizontal(point1, point2)` that takes two points represented by tuples of the form (x, y).
```
[the remark this answers is not in the document]
Comment (jqi26) [reply codewof/programming/content/en/horizontal-line/question.md:3]: Added the note

File: codewof/programming/content/en/is-a-divisor/solution.py [hunk 6]
```diff
@@ -0,0 +1,6 @@
+initial_number = int(input("Number: "))
+number = int(input("Number: "))
+while number != 0:
+    is_divisor = initial_number % number == 0
+    print(f"{number} is a divisor" if is_divisor else f"{number} is not a divisor")
```
[the remark this answers is not in the document]
Comment (jqi26) [reply codewof/programming/content/en/is-a-divisor/solution.py:5]: I was wondering what the solutions were for. Done.

File: codewof/programming/content/en/largest-number/solution.py [hunk 7]
```diff
@@ -0,0 +1,7 @@
+numbers = []
+
+for _ in range(5):
```
[the remark this answers is not in the document]
Comment (jqi26) [reply codewof/programming/content/en/largest-number/solution.py:3]: Done

File: codewof/programming/content/en/quadratic-roots/question.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+#Quadratic Roots
+
+Write a function `number_of_roots(a, b, c)` that takes the `a`, `b`, and `c` coefficients of a quadratic equation and returns the number of real roots.
```
[the remark this answers is not in the document]
Comment (jqi26) [reply codewof/programming/content/en/quadratic-roots/question.md:3]: Removed the question

File: codewof/programming/content/en/simple-calculator/solution.py [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+    if operator == '+':
+        return x + y
+    if operator == '-':
+        return x - y
+    if operator == 'x':
+        return x * y
+    if operator == '/':
+        return x // y
+
+    return False
```
[the remark this answers is not in the document]
Comment (jqi26) [reply codewof/programming/content/en/simple-calculator/solution.py:11]: Got rid of the last return.

File: codewof/programming/content/en/vertical-line/question.md (hunk 1 above)
Comment (jqi26) [reply codewof/programming/content/en/vertical-line/question.md:5]: Done

File: codewof/programming/content/structure/questions.yaml (hunk 2 above)
Comment (jqi26) [reply codewof/programming/content/structure/questions.yaml:1023]: Got rid of the last extra line

**Pull request merged**